### PR TITLE
[ITS3, ITS] Add possibility to switch ITS3 version with configKeyValues

### PR DESCRIPTION
**Pull request by fgrosa**

### CODE REVIEW

File: Detectors/ITSMFT/ITS/simulation/src/Detector.cxx [hunk 1]
```diff
@@ -145,7 +145,10 @@ Detector::Detector(Bool_t active, TString name)
     mDescriptorIB.reset(new DescriptorInnerBarrelITS2(3));
   } else if (name == "IT3") {
 #ifdef ENABLE_UPGRADES
-    mDescriptorIB.reset(new DescriptorInnerBarrelITS3(DescriptorInnerBarrelITS3::ThreeLayersNoDeadZones));
+    mDescriptorIB.reset(new DescriptorInnerBarrelITS3());
+    if (its3Version != "") {
+      dynamic_cast<DescriptorInnerBarrelITS3*>(mDescriptorIB.get())->setVersion(its3Version.Data());
```
Comment (mconcas) [inline Detectors/ITSMFT/ITS/simulation/src/Detector.cxx]: Out of curiosity, why a `dynamic_cast` here?

**Review by mconcas — DISMISSED** [review]

File: Detectors/ITSMFT/ITS/simulation/src/Detector.cxx (hunk 1 above)
Comment (fgrosa) [reply Detectors/ITSMFT/ITS/simulation/src/Detector.cxx]: There is no specific reason indeed, so I changed it (and the others) and tested it that it works properly using simple casts as well.

**Review by iouribelikov — DISMISSED** [review]

**Comment by mconcas** [comment]
@shahor02 error seems unrelated and not genuine, if you could merge, pls.

**Review by mconcas — DISMISSED** [review]

**Comment by fgrosa** [comment]
@shahor02 all the checks are now green and @mconcas and @iouribelikov approved already, so can it be merged? Many thanks!

**Pull request merged**